packages/client/components/DashNavList/DashNavList.tsx

```diff
@@ -41,27 +40,47 @@ const DashHR = styled('div')({
   width: 'calc(100% + 8px)'
 })
 
+const StyledLeftDashNavItem = styled(LeftDashNavItem)<{isViewerOnTeam: boolean}>(
+  ({isViewerOnTeam}) => ({
+    color: isViewerOnTeam ? PALETTE.SLATE_700 : PALETTE.SLATE_600
+  })
+)
+
 interface Props {
   className?: string
-  viewer: DashNavList_viewer$key | null
+  organizationsRef: DashNavList_organization$key | null
   onClick?: () => void
 }
 
-type Team = DashNavList_viewer$data['teams'][0]
+type Team = DashNavList_organization$data[0]['allTeams'][0]
 
 const DashNavList = (props: Props) => {
-  const {className, onClick, viewer: viewerRef} = props
-  const viewer = useFragment(
+  const {className, onClick, organizationsRef} = props
+  const organizations = useFragment(
     graphql`
-      fragment DashNavList_viewer on User {
-        teams {
+      fragment DashNavList_organization on Organization @relay(plural: true) {
+        allTeams {
+          ...DashNavListTeam @relay(mask: false)
+        }
+        viewerTeams {
           ...DashNavListTeam @relay(mask: false)
         }
+        viewerOrganizationUser {
+          user {
+            featureFlags {
+              publicTeams
+            }
+          }
+        }
       }
     `,
-    viewerRef
+    organizationsRef
+  )
+  const hasPublicTeamsFlag =
+    organizations?.[0]?.viewerOrganizationUser?.user?.featureFlags?.publicTeams
+  const teams = organizations?.flatMap((org) =>
+    hasPublicTeamsFlag ? org.allTeams : org.viewerTeams
```

Comment:
If I randomly add this flag to some user, and for example this user has 3 organizations. Now he can see the entire list of teams in those three organizations and send them requests to join without permission from those organizations. 

Moreover, he will also automatically have access to the team list in all future organizations where he can be added afterwards. That is, even if we only select users with one particular organization, he can join new ones after adding the feature flag and we have no control over it. 

Would such requests surprise the organizations and raise security concerns? Should we use the org level feature flag and notify organizations allowing them opt-out?

Reply:
Good point - I was thinking it would be inconsistent to see a different UI for some orgs vs others, but I agree that if an org dislikes public teams, it would be easier to to disable an org feature flag